From 2e9bda15cd49e21894c599492fe60f5b5552d02e Mon Sep 17 00:00:00 2001
From: Arnd Bergmann <arnd@arndb.de>
Date: Fri, 2 Feb 2024 13:50:56 +0100
Subject: [PATCH 01/15] NOTCBL: FROMLIST: ntfs3: avoid an unused variable
 warning

A newly introduced variable is only referenced in an #ifdef:

fs/ntfs3/frecord.c: In function 'ni_read_frame':
fs/ntfs3/frecord.c:2460:16: error: unused variable 'i_size' [-Werror=unused-variable]

Move it into the same conditional block.

Fixes: 4fd6c08a16d7 ("fs/ntfs3: Use i_size_read and i_size_write")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Tested-by: Guenter Roeck <linux@roeck-us.net>
Link: https://lore.kernel.org/r/20240202125101.986302-1-arnd@kernel.org
---
 fs/ntfs3/frecord.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/ntfs3/frecord.c b/fs/ntfs3/frecord.c
index 3b42938a9d3b..7f27382e0ce2 100644
--- a/fs/ntfs3/frecord.c
+++ b/fs/ntfs3/frecord.c
@@ -2457,7 +2457,6 @@ int ni_read_frame(struct ntfs_inode *ni, u64 frame_vbo, struct page **pages,
 	struct ATTR_LIST_ENTRY *le = NULL;
 	struct runs_tree *run = &ni->file.run;
 	u64 valid_size = ni->i_valid;
-	loff_t i_size = i_size_read(&ni->vfs_inode);
 	u64 vbo_disk;
 	size_t unc_size;
 	u32 frame_size, i, npages_disk, ondisk_size;
@@ -2509,6 +2508,7 @@ int ni_read_frame(struct ntfs_inode *ni, u64 frame_vbo, struct page **pages,
 		err = -EOPNOTSUPP;
 		goto out1;
 #else
+		loff_t i_size = i_size_read(&ni->vfs_inode);
 		u32 frame_bits = ni_ext_compress_bits(ni);
 		u64 frame64 = frame_vbo >> frame_bits;
 		u64 frames, vbo_data;

base-commit: 1c892cdd8fe004ed6cef4501a7141594a1616368
-- 
2.43.2

